byterestclient/__init__.py

```diff
@@ -35,12 +35,19 @@ def request(self, method, path, data=None, *args, **kwargs):
             **kwargs
         )
 
-        response.raise_for_status(allow_redirects=False)
+        response.raise_for_status()
+
+        if 300 <= response.status_code < 400:
+            # support 0.12.1 by setting response seperately
+            exception = HTTPError('%s Encountered Redirect' % response.status_code)
+            exception.response = response
+            raise exception
```

Comment:
I will not implement this after all, @zoni, because 304 will complicate
things to no end. It has no response content, so it is not deserializable.
Also:

```
If the client has performed a conditional GET request... the server SHOULD
respond with this status code.
```

We never do conditional requests (Last-Modified, ETag, If-Modified-Since,
If-None-Match) never come into play; the server never sends Age nor ETag,
nor Vary.